### ISSUE TITLE
Privoxy not starting

### ISSUE THREAD

**Issue by stueja** [freedombox/Plinth#464]
Privoxy installs correctly, but later the service cannot be started. Diagnosis shows that binding to ports 8118 (ipv4/v6) doesn't work. 
```
root@fbx:/# systemctl status privoxy
● privoxy.service - Privacy enhancing HTTP Proxy
   Loaded: loaded (/lib/systemd/system/privoxy.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2016-05-18 15:12:27 UTC; 2h 28min ago
     Docs: man:privoxy(8)
           https://www.privoxy.org/user-manual/
  Process: 359 ExecStart=/usr/sbin/privoxy --pidfile $PIDFILE --user $OWNER $CONFIGFILE (code=exited, status=1/FAILURE)

May 18 15:12:26 fbx systemd[1]: Starting Privacy enhancing HTTP Proxy...
May 18 15:12:27 fbx systemd[1]: privoxy.service: Control process exited, co...=1
May 18 15:12:27 fbx systemd[1]: Failed to start Privacy enhancing HTTP Proxy.
May 18 15:12:27 fbx systemd[1]: privoxy.service: Unit entered failed state.
May 18 15:12:27 fbx systemd[1]: privoxy.service: Failed with result 'exit-code'.
Hint: Some lines were ellipsized, use -l to show in full.
```
```
root@fbx:/# cat /var/log/privoxy/logfile
2016-05-18 15:05:47.713 b6fa6390 Fatal error: can't bind to ::1:8118: There may be another Privoxy or some other proxy running on port 8118
2016-05-18 15:12:26.890 b6f06390 Fatal error: can't bind to ::1:8118: There may be another Privoxy or some other proxy running on port 8118
```
```
root@fbx:/# netstat -tulpen
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode      PID/Program name
tcp        0      0 0.0.0.0:9050            0.0.0.0:*               LISTEN      0          14768      614/tor
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      0          14970      847/apache2
tcp        0      0 127.0.0.1:9051          0.0.0.0:*               LISTEN      0          14773      614/tor
tcp        0      0 0.0.0.0:38300           0.0.0.0:*               LISTEN      0          14778      614/tor
tcp        0      0 127.0.0.1:8000          0.0.0.0:*               LISTEN      0          17626      464/python3
tcp        0      0 0.0.0.0:389             0.0.0.0:*               LISTEN      0          13246      540/slapd
tcp        0      0 127.0.0.1:42251         0.0.0.0:*               LISTEN      0          14779      614/tor
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      0          14968      847/apache2
tcp        0      0 127.0.0.1:9040          0.0.0.0:*               LISTEN      0          14771      614/tor
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      0          15897      469/sshd
tcp6       0      0 :::9050                 :::*                    LISTEN      0          14767      614/tor
tcp6       0      0 :::44517                :::*                    LISTEN      113        14808      753/obfs4proxy
tcp6       0      0 :::389                  :::*                    LISTEN      0          13247      540/slapd
tcp6       0      0 ::1:9040                :::*                    LISTEN      0          14772      614/tor
tcp6       0      0 :::34837                :::*                    LISTEN      113        14813      753/obfs4proxy
tcp6       0      0 :::22                   :::*                    LISTEN      0          15899      469/sshd
udp        0      0 0.0.0.0:1194            0.0.0.0:*                           0          12104      342/openvpn
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           106        11803      329/avahi-daemon: r
udp        0      0 0.0.0.0:68              0.0.0.0:*                           0          15090      826/dhclient
udp        0      0 127.0.0.1:9053          0.0.0.0:*                           0          14769      614/tor
udp        0      0 0.0.0.0:48238           0.0.0.0:*                           106        11805      329/avahi-daemon: r
udp        0      0 192.168.122.59:123      0.0.0.0:*                           107        15200      510/ntpd
udp        0      0 10.91.0.1:123           0.0.0.0:*                           0          13203      510/ntpd
udp        0      0 127.0.0.1:123           0.0.0.0:*                           0          13201      510/ntpd
udp        0      0 0.0.0.0:123             0.0.0.0:*                           0          13197      510/ntpd
udp6       0      0 :::59085                :::*                                106        11806      329/avahi-daemon: r
udp6       0      0 :::5353                 :::*                                106        11804      329/avahi-daemon: r
udp6       0      0 ::1:9053                :::*                                0          14770      614/tor
udp6       0      0 fe80::a2f6:fdff:fe8:123 :::*                                107        15204      510/ntpd
udp6       0      0 ::1:123                 :::*                                0          13205      510/ntpd
udp6       0      0 :::123                  :::*                                0          13194      510/ntpd

```

**Comment by jvalleroy** [comment]
I can confirm this issue.

The default privoxy config from Debian has 2 listen-address lines now:
```
listen-address  127.0.0.1:8118
listen-address  [::1]:8118
```

Our setup script changes the 1st line to [::]:8118, which overlaps the 2nd line. We can either have the setup script remove the extra line, or use debconf preseeding to set privoxy/listen-address.

**Comment by SunilMohanAdapa** [comment]
On 05/24/2016 08:06 PM, James Valleroy wrote:
> I can confirm this issue.
> 
> The default privoxy config from Debian has 2 listen-address lines now:
> 
> |listen-address 127.0.0.1:8118 listen-address [::1]:8118 |
> 
> Our setup script changes the 1st line to [::]:8118, which overlaps the
> 2nd line. We can either have the setup script remove the extra line, or
> use debconf preseeding to set privoxy/listen-address.
> 

This should now fixed with #467.

-- 
Sunil
